app/views/mailers/notify_mailer/new_badge_email.html.erb

```diff
@@ -8,23 +8,29 @@
   <p>
     <%= @badge.description %>
   </p>
+
+  <% if @badge_achievement.rewarding_context_message %>
```

Comment:
I suggest using `@badge_achievement.rewarding_context_message?` to avoid displaying an empty paragraph when the context message is empty (though there are no such cases right now).
+ indentation is missing :)

Reply:
I'll add the indentation, but I didn't get the other note, that's what the code is doing, doesn't it?

Reply:
I mean if the `@badge_achievement.rewarding_context_message` is equal to an empty string, the empty `<p>`, `<em>` and `<br />` will be rendered. If we change `if @badge_achievement.rewarding_context_message` to `if @badge_achievement.rewarding_context_message?` and the `@badge_achievement.rewarding_context_message` is equal to an empty string, empty tags and the br won't be rendered.

Reply:
oooh, I got it! You're right, the text checks for `nil`, but not empty string. I'll fix it :)